File: backend/src/main/java/com/portfolio/entity/Skill.java
package com.portfolio.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "skills")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Skill {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable=false,length=80)
    private String name;

    @Column(nullable=false,length=60)
    private String category;

    @Column(nullable=false)
    private Integer proficiency;

    @Column(name="icon_name",length=80)
    private String iconName;

    @Column(name="sort_order")
    @Builder.Default
    private Integer  sortOrder = 0;

    @Builder.Default
    private Boolean active = true;
}

File: frontend/src/types/skill.ts
export interface Skill {
  id: number;
  name: string;
  category?: string; // e.g., "Frontend", "Backend", "DevOps"
   proficiency?: number; // e.g., 1-5 scale
   iconName?: string; // e.g., "react", "nodejs", "docker"
   sortOrder?: number; // for ordering skills in a list
}